From e74439d0a3a12b7c90826dd4f8bded6cbe0e142a Mon Sep 17 00:00:00 2001
Message-ID: <e74439d0a3a12b7c90826dd4f8bded6cbe0e142a.1748904529.git.osandov@osandov.com>
From: Omar Sandoval <osandov@osandov.com>
Date: Mon, 2 Jun 2025 15:26:09 -0700
Subject: [PATCH] page_pool: enable CONFIG_PAGE_POOL by default

Since commit b530e9e1063e ("bpf: Add "live packet" mode for XDP in
BPF_PROG_RUN") (in v5.18), CONFIG_BPF_SYSCALL selects CONFIG_PAGE_POOL.
Before that, only certain network drivers enabled it. In order to test
page pool drgn helpers without enabling unnecessary network drivers,
always enable CONFIG_PAGE_POOL.

Signed-off-by: Omar Sandoval <osandov@osandov.com>
---
 net/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/Kconfig b/net/Kconfig
index 8a1f9d0287de..865f93e259da 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -433,6 +433,7 @@ config NET_DEVLINK
 
 config PAGE_POOL
 	bool
+	default y
 
 config FAILOVER
 	tristate "Generic failover module"
-- 
2.49.0

